s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/SceneRepositoryImpl.java

```diff
@@ -0,0 +1,154 @@
+package pl.cyfronet.s4e.data.repository;
+
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.apache.commons.dbutils.handlers.BeanListHandler;
+import pl.cyfronet.s4e.bean.Scene;
+import pl.cyfronet.s4e.controller.request.ZoneParameter;
+import pl.cyfronet.s4e.properties.DbProperties;
+import pl.cyfronet.s4e.util.TimeHelper;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.time.*;
+import java.util.List;
+import java.util.Map;
+
+public class SceneRepositoryImpl implements SceneRepositoryExt {
+    private final DbProperties dbProperties;
+    private ResultSetHandler<List<Scene>> handler;
+    private QueryRunner runner;
+    private Connection conn;
+
+    public SceneRepositoryImpl(DbProperties dbProperties, TimeHelper timeHelper) throws SQLException {
+        this.dbProperties = dbProperties;
+        this.handler = new BeanListHandler<>(Scene.class, new SceneHandler());
+        this.runner = new QueryRunner();
+        this.conn = DriverManager.getConnection(dbProperties.getUrl(), dbProperties.getUsername(), dbProperties.getPassword());
+    }
+
+    public List<Scene> findAllByParamsMap(Map<String, Object> params) throws SQLException {
+        return runner.query(conn, prepareQuery(params), handler);
+    }
+
+    private String prepareQuery(Map<String, Object> params) {
+        StringBuilder resultQuery = new StringBuilder();
+        resultQuery.append("SELECT s.id, s.product_id FROM Scene s WHERE true");
```

Comment:
Wouldn't it be better to use some query DSL for that? I'm not sure we're protected from query injection the way it is done now. I'd test against it, anyway.